src/main/java/ui/ConfirmChangesDialog.java

```diff
@@ -0,0 +1,26 @@
+package ui;
+
+import javafx.scene.control.Alert;
+import javafx.scene.control.ButtonType;
+import javafx.stage.Stage;
+
+/**
+ * This class represents a confirm dialog that asks the user whether or not to save the current board before
+ * switching to another board, or creating a new board.
+ */
+public class ConfirmChangesDialog extends Alert {
+
+    private static final String DIALOG_TITLE = "Save Changes?";
+    private static final String DIALOG_HEADER = "There are unsaved changes to your current board.";
+    private static final String DIALOG_CONTENT = "If you choose not to save, all unsaved changes will be discarded.\n\n"
+                                               + "Do you want to save them?";
+
+    public ConfirmChangesDialog(Stage stage) {
+        super(AlertType.CONFIRMATION);
```

Comment:
Since you use `confirmation` type here. I think there will be some UI
difference, also. Unless if your dialog is also a warning.
On Apr 6, 2016 5:24 PM, "Liu Xinan" <notifications@github.com> wrote:

> In src/main/java/ui/ConfirmChangesDialog.java
> <https://github.com/HubTurbo/HubTurbo/pull/1485#discussion_r58675841>:
>
> > +import javafx.scene.control.ButtonType;
> > +import javafx.stage.Stage;
> > +
> > +/**
> > + * This class represents a confirm dialog that asks the user whether or not to save the current board before
> > + * switching to another board, or creating a new board.
> > + */
> > +public class ConfirmChangesDialog extends Alert {
> > +
> > +    private static final String DIALOG_TITLE = "Save Changes?";
> > +    private static final String DIALOG_HEADER = "There are unsaved changes to your current board.";
> > +    private static final String DIALOG_CONTENT = "If you choose not to save, all unsaved changes will be discarded.\n\n"
> > +                                               + "Do you want to save them?";
> > +
> > +    public ConfirmChangesDialog(Stage stage) {
> > +        super(AlertType.CONFIRMATION);
>
> Why use of the existing showYesNoWarningDialog is not recommended?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/HubTurbo/HubTurbo/pull/1485/files/e736de8ed9249ffc9ffb1cc5cd3c21adfcaf0563#r58675841>
>